nesta/packages/geo_utils/geocode.py

```diff
@@ -113,15 +114,23 @@ def geocode_batch_dataframe(df, city='city', country='country',
         country (str): name of the input column containing the country
         latitude (str): name of the output column containing the latitude
         longitude (str): name of the output column containing the longitude
+        query_method (int): query methods to attempt: 0: city, country only
```

Comment:
Apologies for the delay here in reviewing. I think the 0,1,2 method might leave us vulnerable to difficult debugging in the future. I'd favour using string labels unless performance is an issue - whereas the bottleneck in this case is really the geocode API - so ['city_country_only', 'query_only', 'both'] might be more useful. I don't think that would take long to implement right?

Reply:
This makes sense, it was already annoying to remember what each code meant while I was just changing them over